### PR TITLE
Make nrows/ncols part of required interface for matrix spaces/algebras.

### PR DESCRIPTION
**Pull request by wbhart**
Fixes https://github.com/Nemocas/AbstractAlgebra.jl/issues/965

### CODE REVIEW

**Comment by rfourquet** [comment]
If `nrows` becomes part of the required interface for matrix space, maybe it could become optional for matrices? The fallback defintion could be `nrows(matrix) = nrows(parent(matrix))`, although it would be slow in some cases, in particular when the parent is not in the field of the matrix and is created on demand.

**Comment by wbhart** [comment]
That makes sense to me.

File: src/Matrix.jl [hunk 1]
```diff
@@ -91,6 +91,20 @@ check_square(S::MatAlgebra) = S
...
+Return the number of rows of the given matrix space.
+"""
+nrows(a::MatSpace) = a.nrows
+
+@doc Markdown.doc"""
+    ncols(a::MatSpace)
+
+Return the number of columns of the given matrix space.
+"""
+ncols(a::MatSpace) = a.ncols
```
Comment (thofma) [inline src/Matrix.jl:106]: Technically, these should be in `generic/`, right?

**Comment by thofma** [comment]
Do you want to implement this suggestion or should we put this on the todo list?

**Comment by wbhart** [comment]
I can add it I guess.

File: src/Matrix.jl (hunk 1 above)
Comment (wbhart) [reply src/Matrix.jl:106]: I put them there originally and then realised that we may as well make them fallback defaults for people who have an nrows/ncols field in their matrix space object.

**Comment by thofma** [comment]
We can also defer this decision, I am fine with that.

**Comment by wbhart** [comment]
Ok, why don't you go ahead and merge without it and I'll open a ticket to implement it.

**Pull request merged**